File: backend/DTO/Request/CustomerRequest.cs
using System.ComponentModel.DataAnnotations;

namespace backend.DTO.Request
{
    public class CustomerRequest
    {
        [Required]
        public required string FirstName { get; set; }
        [Required]
        public required string LastName { get; set; }
        [Required]
        public required string Email { get; set; }
        [Required]
        public required string Password { get; set; }
        [Required]
        public required string Address { get; set; }
        [Required]
        public required string PhoneNumber { get; set; }
    }

}

File: frontend_client/lib/model/request/customer_request.dart
import 'package:equatable/equatable.dart';

class CustomerRequest extends Equatable {
  const CustomerRequest({
    this.firstName,
    this.lastName,
    this.email,
    this.password,
    this.address,
    this.phoneNumber,
  });

  factory CustomerRequest.fromJson(Map<String, dynamic> json) {
    return CustomerRequest(
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      password: json['password'],
      address: json['address'],
      phoneNumber: json['phoneNumber'],
    );
  }
  final String? firstName;
  final String? lastName;
  final String? email;
  final String? password;
  final String? address;
  final String? phoneNumber;

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'password': password,
      'address': address,
      'phoneNumber': phoneNumber,
    };
  }

  @override
  List<Object?> get props => [
    firstName,
    lastName,
    email,
    password,
    address,
    phoneNumber,
  ];
}
